client/src/modules/dashboard/Events/components/EventFormUtils.ts

```diff
@@ -161,3 +161,83 @@ export const formatValue = (field: Field, store?: IEventData): any => {
 
   return store[key];
 };
+
+export const getAllowedSponsorTypes = (
+  sponsorData: SponsorsQuery,
+  sponsorTypes: EventSponsorTypeInput[],
+  watchSponsorsArray: EventSponsorInput[],
+  sponsorFieldId: number,
+) =>
+  sponsorTypes.filter(({ type }) =>
+    hasTypeAllowedSponsors(
+      sponsorData,
+      type,
+      watchSponsorsArray,
+      sponsorFieldId,
+    ),
+  ) ?? [];
```

Comment:
```suggestion
  sponsorTypes.filter(
    ({ type }) =>
      getAllowedSponsorsForType(
        sponsorData,
        type,
        watchSponsorsArray,
        sponsorFieldId,
      ).length,
  );
```
It feels like overkill to create an extra function when `.length` does the same job.

Reply:
This is lovely, I didn't realize `.length` on its own is already doing enough for `filter`.

Reply:
Yeah, it works because 0 is the only falsy number. Which is pretty neat for things like this.